Python/traceback.c

```diff
@@ -980,9 +980,6 @@ dump_traceback(int fd, PyThreadState *tstate, int write_header)
             /* Trampoline frame */
             frame = frame->previous;
         }
-        if (frame == NULL) {
```

Comment:
Why, but why do you think that ``frame->previous`` on L981 is not ``NULL``?

Reply:
If frame is part of a chain and there might be a situation where it’s traversed backwards, then at that point in the code https://github.com/python/cpython/blob/7a2d77c903f29d7ea08b870b8e3fa2130f667a59/Python/traceback.c#L975
it should break out. That’s what I think.

Reply:
```
static void
dump_traceback(int fd, PyThreadState *tstate, int write_header)
{
    if (write_header) {
        PUTS(fd, "Stack (most recent call first):\n");
    }

    if (tstate_is_freed(tstate)) {
        PUTS(fd, "  <tstate is freed>\n");
        return;
    }

    _PyInterpreterFrame *frame = tstate->current_frame;
    if (frame == NULL) {
        PUTS(fd, "  <no Python frame>\n");
        return;
    }

    unsigned int depth = 0;
    while (1) {
        if (MAX_FRAME_DEPTH <= depth) {
            if (MAX_FRAME_DEPTH < depth) {
                PUTS(fd, "plus ");
                _Py_DumpDecimal(fd, depth);
                PUTS(fd, " frames\n");
            }
            break;
        }
        dump_frame(fd, frame);
        frame = frame->previous;
        if (frame == NULL) {
            break;
        }
        if (frame->owner == FRAME_OWNED_BY_CSTACK) {
            /* Trampoline frame */
            frame = frame->previous;
        }
        // if (frame == NULL) {
        //     break;
        // }
        /* Can't have more than one shim frame in a row */
        assert(frame->owner != FRAME_OWNED_BY_CSTACK);
        depth++;
    }
}
```
I hid the targets I needed to modify by commenting them out.
https://github.com/python/cpython/blob/7a2d77c903f29d7ea08b870b8e3fa2130f667a59/Python/traceback.c#L983-L985

I then ran the tests using `./python Lib\test\test_traceback.py`.
It seemed to run well as it passed all 347 tests in about 8.3 seconds.
```
PS E:\code\cc\cpython\alpha\cpython-main> ./python .\Lib\test\test_traceback.py
Running Debug|x64 interpreter...
...........................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 347 tests in 8.303s

OK
```
But I can't understand why it fails on CI. Maybe I ran the test incorrectly.